include/aspect/initial_conditions/adiabatic_boundary.h

```diff
@@ -0,0 +1,102 @@
+/*
+   Copyright (C) 2015 by the authors of the ASPECT code.
+    
+   This file is part of ASPECT.
+ 
+   ASPECT is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2, or (at your option)
+   any later version.
+  
+   ASPECT is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with ASPECT; see the file doc/COPYING.  If not see
+   <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/initial_conditions/interface.h>
+
+/** This initial condition is designed for the 3D ellipsoid chunk geometry
+ * model. It discretizes the model domain into two regions separated by an
+ * isotherm below which the temperature increases adiabatically. The user
+ * defines the location of the thermal isotherm with a data file with the
+ * format defined in the ASPECT manual. Note that the latitudinal and
+ * longitudinal bounds of the ascii input data file needs to be at least 1
+ * degree wider than the bounds you use to define the ellipsoid chunk geometry
+ * model.
+ * This plugin is developed by Tahiry Rajaonarison, D. Sarah Stamps, and Wolfgang Bangerth.
+ *
+ * Here is class that compute initial temperature field based on a 
+ * user-defined adiabatic boundary. Below the adiabatic boundary the
+ * temperature increases adiabatically while above the adiabatic 
+ * boundary the temperature linearly increases from a surface temperature
+ * (273.15 K or 0 degree C) to an isotherm (1673.15 K or 1600 degree C)
+ * that is the adiabatic boundary.
```

Comment:
I meant to move this documentation to the *class declaration*, not to the namespace declaration. I think it would also be useful if you *merged* the documentation, not just *collated* it.